api-gen/rendering/renderables.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ClassEntry,
+  ConstantEntry,
+  DirectiveEntry,
+  DocEntry,
+  EnumEntry,
+  EnumMemberEntry,
+  FunctionEntry,
+  JsDocTagEntry,
+  MemberEntry,
+  ParameterEntry,
+  PipeEntry,
+} from './entities';
+
+/** JsDoc tag info augmented with transformed content for rendering. */
+export interface JsDocTagRenderable extends JsDocTagEntry {
+  htmlComment: string;
+}
+
+/** A documentation entry augmented with transformed content for rendering. */
+export interface DocEntryRenderable extends DocEntry {
+  htmlDescription: string;
+  jsdocTags: JsDocTagRenderable[];
+}
+
+/** Documentation entity for a constant augmented transformed content for rendering. */
+export type ConstantEntryRenderable = ConstantEntry & DocEntryRenderable;
+
+/** Documentation entity for a TypeScript class augmented transformed content for rendering. */
+export type ClassEntryRenderable = ClassEntry &
+  DocEntryRenderable & {
+    members: MemberEntryRenderable[];
+  };
+
+/** Documentation entity for a TypeScript enum augmented transformed content for rendering. */
+export type EnumEntryRenderable = EnumEntry &
+  DocEntryRenderable & {
+    members: EnumMemberEntryRenderable[];
+  };
+
+/**
+ * Documentation entity for an Angular directives and components augmented transformed
+ * content for rendering.
+ */
+export type DirectiveEntryRenderable = DirectiveEntry & DocEntryRenderable;
+
+/** Documentation entity for a pipe augmented transformed content for rendering. */
+export type PipeEntryRenderable = PipeEntry & DocEntryRenderable;
+
+export type FunctionEntryRenderable = FunctionEntry &
+  DocEntryRenderable & {
+    params: ParameterEntryRenderable[];
+  };
+
+/** Sub-entry for a single class or enum member augmented with transformed content for rendering. */
+export interface MemberEntryRenderable extends MemberEntry {
+  htmlDescription: string;
+  jsdocTags: JsDocTagRenderable[];
+}
+
+/** Sub-entry for an enum member augmented transformed content for rendering. */
+export type EnumMemberEntryRenderable = EnumMemberEntry & MemberEntryRenderable;
+
+/** Sub-entry for a class property augmented transformed content for rendering. */
+export type PropertyEntryRenderable = MemberEntryRenderable;
+
+/** Sub-entry for a class method augmented transformed content for rendering. */
+export type MethodEntryRenderable = MemberEntryRenderable &
+  FunctionEntryRenderable & {
+    params: ParameterEntryRenderable[];
+  };
+
+/** Sub-entry for a single function parameter augmented transformed content for rendering. */
+export interface ParameterEntryRenderable extends ParameterEntry {
+  htmlDescription: string;
+}
+
+/** A doc entry that has jsdoc tags. */
+export interface HasJsDocTags {
```

Comment:
Wondering if we should move all these `HasX` interfaces to a separate file and only have the actual renderable entities here?

Reply:
I was on the fence on that, too. I've now split them out to a separate file (`traits.ts`) and then actually moved that plus a few other related files to a new `entities` directory under `rendering`.